docs/other_vscode.md

```diff
@@ -117,3 +117,77 @@ Using the [standard `compile_commands.json` database](https://clang.llvm.org/doc
 1. Start typing `clangd: Restart Language Server` and select it when it appears.
 
 Now you're ready to code QMK Firmware in VS Code!
+
+And now you're ready to code QMK Firmware in VS Code!
+
+# Debugging ARM MCUs with Visual Studio Code
+
+**...and a Black Magic Probe.**
+
+Visual Studio Code has the ability to debug applications, but requires some configuration in order to get it to be able to do so for ARM targets.
+
+This documentation describes a known-working configuration for setting up the use of a Black Magic Probe to debug using VS Code.
+
+It is assumed that you've correctly set up the electrical connectivity of the Black Magic Probe with your MCU. Wiring up `NRST`, `SWDIO`, `SWCLK`, and `GND` should be enough.
+
+Install the following plugin into VS Code:
+
+* [Cortex-Debug](https://marketplace.visualstudio.com/items?itemName=marus25.cortex-debug) - 
+  This adds debugger support for ARM Cortex targets to VS Code.
+
+A debugging target for the MCU for your board needs to be defined, and can be done so by adding the following to a `.vscode/launch.json` file:
+
+```json
+{
+  // Use IntelliSense to learn about possible attributes.
+  // Hover to view descriptions of existing attributes.
+  // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
+  "version": "0.2.0",
+  "configurations": [
+    {
+      "name": "Black Magic Probe (OneKey Proton-C)",
+      "type": "cortex-debug",
+      "request": "launch",
+      "cwd": "${workspaceRoot}",
+      "executable": "${workspaceRoot}/.build/handwired_onekey_proton_c_default.elf",
+      "servertype": "bmp",
+      "BMPGDBSerialPort": "COM4",
+      "svdFile": "Q:\\svd\\STM32F303.svd",
+      "device": "STM32F303",
+      "v1": false,
+      "windows": {
+        "armToolchainPath": "Q:\\gcc-arm-none-eabi-9-2020-q2-update-win32\\bin"
+      }
+    }
+  ]
+}
+```
+
+You'll need to perform some modifications to the file above in order to target your specific device:
+
+* `"name"`: Can be anything, but if you're debugging multiple targets you'll want something descriptive here.
+* `"cwd"`: The path to the QMK Firmware repository root directory -- _if using the `.vscode` directory existing in the `qmk_firmware` git repository, the default above should be correct_
+* `"executable"`: The path to the `elf` file generated as part of the build for your keyboard -- _exists in `<qmk_firmware>/.build`_
+* `"BMPGDBSerialPort"`: The `COM` port under Windows, or the `/dev/...` path for Linux/macOS.
```

Comment:
messing with this, it seems that it creates 2 serial ports, for me.  Might be worth mentioning that, and to use the first?

Reply:
Done.